Packs/CommonPlaybooks/Playbooks/playbook-IP_Enrichment_-_Generic_v2.yml

```diff
@@ -196,14 +273,301 @@ tasks:
+    type: regular
+    task:
+      id: 4b707093-e557-49e6-8825-f7b23c1f12a6
+      version: -1
+      name: Determine whether the IP address is internal or external
+      description: "Checks if the IP address is in the internal IP address ranges.\nIf internal, sets the \"InRange\" attribute to \"yes\".\nIf external, sets the \"InRange\" attribute to \"no\". "
+      scriptName: IsIPInRanges
+      type: regular
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "34"
+    scriptarguments:
+      ip:
+        complex:
+          root: inputs.IP
+          transformers:
+          - operator: uniq
+      ipRanges:
+        complex:
+          root: inputs.InternalRange
+    reputationcalc: 1
+    separatecontext: false
+    continueonerrortype: ""
+    view: |-
+      {
+        "position": {
+          "x": 850,
+          "y": 530
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "28":
+    id: "28"
+    taskid: 6c306aba-243f-4d13-888c-a142fcf86737
+    type: condition
+    task:
+      id: 6c306aba-243f-4d13-888c-a142fcf86737
+      version: -1
+      name: Can common logic be executed from the parent playbook?
+      description: Checks whether the playbook inputs specify that common logic like the classification of IP addresses to ranges and resolving them should be executed in the main (parent) enrichment playbook, instead of in the sub-playbooks.
+      type: condition
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#default#':
+      - "24"
+      No, execute in sub-playbooks:
+      - "34"
+      "Yes":
+      - "33"
+    separatecontext: false
+    conditions:
+    - label: "Yes"
+      condition:
+      - - operator: isEqualString
+          left:
+            value:
+              complex:
+                root: inputs.ExecutedFromParent
+            iscontext: true
+          right:
+            value:
+              simple: "True"
+          ignorecase: true
+      - - operator: isNotEmpty
+          left:
+            value:
+              complex:
+                root: inputs.IP
+            iscontext: true
+    - label: No, execute in sub-playbooks
+      condition:
+      - - operator: isNotEmpty
+          left:
+            value:
+              complex:
+                root: inputs.IP
+            iscontext: true
+    continueonerrortype: ""
+    view: |-
+      {
+        "position": {
+          "x": 265,
+          "y": 220
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "29":
+    id: "29"
+    taskid: be9d391d-3019-40bc-85da-e72e9e6813a8
+    type: title
+    task:
+      id: be9d391d-3019-40bc-85da-e72e9e6813a8
+      version: -1
+      name: Resolve IPs
```

Comment:
I wanted to but that would cause over complication and increase the playbook size. I wanted to keep only common logic in the parent playbook, and endpoint enrichment is only specific to internal IP. I didn't like the fact that it was in IP enrichment to begin with :P